source/MRMesh/MRMeshCollide.cpp

```diff
@@ -289,9 +306,19 @@ std::vector<FaceFace> findSelfCollidingTriangles( const MeshPart & mp )
 
                 }
             );
+            
+            if ( sb && std::this_thread::get_id() == mainThreadId )
+            {
+                if ( !reportProgress( sb, float( is - minId ) / float( maxId - minId ) ) )
+                    keepGoing.store( false, std::memory_order_relaxed );
+            }
+
             subtaskRes[is] = std::move( myRes );
         }
-    } );
+    }, tbb::static_partitioner() );
```

Comment:
`static_partitioner` decreases performance too much. It is better to keep default partitioner, and introduce an atomic `numDone` that is updated at the end of range by each thread, and report `(numDone+is-minId)/total`